### ISSUE TITLE
Can't render with GPU

### ISSUE THREAD

**Issue by hoopahmadness** [kochd/electricsheep-hd-client#67]
I'm running daemon in POP_OS! CPU rendering works OK ( I seem to be hampered by constant internal server errors) but I'm able to crank out frames. However, running with --gpu gives me the following output:
```
[15:39:36] INFO: Requesting new work...
[15:39:36] INFO: Solving PoW...
./daemon:126: warning: URI.escape is obsolete
[15:39:40] INFO: Solved after 4 seconds
./daemon:126: warning: URI.escape is obsolete
./daemon:126: warning: URI.escape is obsolete
[15:39:43] INFO: Rendering... electricsheep.244.00976.08574_electricsheep.244.00977.12832 Frame 1054-1054
Only created 1 renderers out of 0 requested, exiting.
Traceback (most recent call last):
	4: from ./daemon:353:in '< main >'
	3: from ./daemon:353:in 'loop'
	2: from ./daemon:397:in 'block in '< main >'
	1: from ./daemon:209:in 'request_work'
./daemon:276:in 'ember_render_frame': Emberrender exited: 1 (RuntimeError)
```
I assume that since nobody else has reported this issue then I might be doing something wrong; can anyone give me a clue?

Edit: sorry, for the life of me I cannot figure out the formatting.

**Comment by hoopahmadness** [comment]
The internal service errors were probably related to me not being a trusted member; now that I'm trusted I'm not seeing any more of those. Please ignore that.

**Comment by hoopahmadness** [comment]
Also just noticed that my GPU is not showing up; I am willing to bet that's related

`$ ./daemon --gpu-list-devices

OpenCL Info: `

**Comment by kochd** [comment]
Looks like your OpenCL is not setup properly. This is not related to this project by any means. What's the vendor of your GPU?
You have to get `clinfo` to list at least one supported device

**Comment by kochd** [comment]
> I cannot figure out the formatting.

Wrap it inside ```

**Comment by hoopahmadness** [comment]
I went down a rabbit hole and established that you're correct; it's an issue with my installation that's unrelated to this project.

My GPU is by AMD and I'm using amdgpu; I'm not really willing to change to amdgpupro. Unfortunately when I install `mesa-opencl-icd` it breaks `clinfo` due to a known bug found here: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=964973

So unless someone here knows of a way around this bug (there's lots of talk about the problem but I haven't seen any steps for me to take to fix it) then I'm happy to close this.

**Comment by kochd** [comment]
No amdgpupro needed. Get rocm up and running: https://github.com/RadeonOpenCompute/ROCm#Deploying-ROCm

mesa-opencl-icd wont solve your problem as clover (mesa's opencl implementation) only supports opencl up to 1.2 where >2.0 is needed here.

**Comment by hoopahmadness** [comment]
Thanks for the tip. Sorry for the delay, I got distracted with the prospect of getting my fans running on a PID controller. 

I've followed the directions for Ubuntu (close enough I assume) but I'm encountering an error. It looks like it doesn't like me being on the 5.8 kernel

```Setting up rock-dkms (1:4.0-23) ...
Loading new amdgpu-4.0-23 DKMS files...
Building for 5.8.0-7630-generic
Building for architecture amd64
Building initial module for 5.8.0-7630-generic
ERROR (dkms apport): kernel package linux-headers-5.8.0-7630-generic is not supp
orted
Error! Bad return status for module build on kernel: 5.8.0-7630-generic (amd64)
Consult /var/lib/dkms/amdgpu/4.0-23/build/make.log for more information.
dpkg: error processing package rock-dkms (--configure):
 installed rock-dkms package post-installation script subprocess returned error 
exit status 10
Setting up llvm-amdgpu (12.0.dev) ...
Setting up rocm-smi (1.0.0-206-rocm-rel-4.0-23-ge39c0e2) ...```

I'll look around.

**Comment by kochd** [comment]
> Note: These directions may not work as written on unsupported Debian-based distributions. For example, newer versions of Ubuntu may not be compatible with the rock-dkms kernel driver. In this case, you can exclude the rocm-dkms and rock-dkms packages.

kfd provided by the dkms packages has been merge into the kernel so the dkms package are no longer needed if you run a more recent kernel.
Your target is to find something like this after boot:
``` shell
# dmesg | grep kfd
[    7.490041] kfd kfd: Allocated 3969056 bytes on gart
[    7.491015] kfd kfd: added device 1002:67df
```

**Comment by kochd** [comment]
```
# dpkg -l | grep rocm | grep ^ii
ii  comgr                                           1.9.0.194-rocm-rel-3.9-17-0fa438b          amd64        Library to provide support functions
ii  hsa-rocr-dev                                    1.2.30900.0-rocm-rel-3.9-17-75f9b74a       amd64        AMD Heterogeneous System Architecture HSA - Linux HSA Runtime for Boltzmann (ROCm) platforms
ii  procmail                                        3.22-26                                    amd64        Versatile e-mail processor
ii  rocm-clang-ocl                                  0.5.0.64-rocm-rel-3.9-17-50fb51a           amd64        OpenCL compilation with clang compiler.
ii  rocm-cmake                                      0.3.0.153-rocm-rel-3.9-17-1d1caa5          amd64        rocm-cmake built using CMake
ii  rocm-dbgapi                                     0.36.0-rocm-rel-3.9-17                     amd64        Library to provide AMD GPU debugger API
ii  rocm-dev                                        3.9.0.30900-17                             amd64        Radeon Open Compute (ROCm) Runtime software stack
ii  rocm-device-libs                                1.0.0.637-rocm-rel-3.9-17-db8c0c3          amd64        Radeon Open Compute - device libraries
ii  rocm-gdb                                        9.2-rocm-rel-3.9-17                        amd64        ROCgdb
ii  rocm-opencl                                     3.6Beta-14-g0c40e05-rocm-rel-3.9-17        amd64        OpenCL: Open Computing Language on ROCclr
ii  rocm-opencl-dev                                 3.6Beta-14-g0c40e05-rocm-rel-3.9-17        amd64        OpenCL: Open Computing Language on ROCclr
ii  rocm-smi                                        1.0.0-206-rocm-rel-3.9-17-ge39c0e2         amd64        System Management Interface for ROCm
ii  rocm-smi-lib64                                  2.7.0.5-rocm-rel-3.9-17-8f9f943            amd64        AMD System Management libraries
ii  rocm-utils                                      3.9.0.30900-17                             amd64        Radeon Open Compute (ROCm) Runtime software stack
ii  rocminfo                                        1.30900.0                                  amd64        Radeon Open Compute (ROCm) Runtime rocminfo tool

```

**Comment by hoopahmadness** [comment]
I got all the same output as you, up until # rocminfo; do you have /opt/ in your path?

**Comment by hoopahmadness** [comment]
This is after I went ahead and apt remove'd the faulty rock-dkms and rocm-dkms packages. `daemon --gpu-list-devices` still doesn't show my card.

**Comment by kochd** [comment]
- Is `daemon --gpu-list-devices` showing nothing then?
- The output just refers to the same information provided by `clinfo` which shows your system's opencl configuration
So if you got the kfd in your dmesg and rocminfo shows your card rocm should be able to utilize it. However if `clinfo` doesn't show your card there could be problem with the icd. 

Maybe this helps? https://github.com/RadeonOpenCompute/ROCm/issues/1131

**Comment by hoopahmadness** [comment]
That's correct, gpu-list-devices still showed nothing.

I followed that link and fiddled around a bit. Not only did nothing work but I've probably uninstalled some necessary driver. I can't play games, my monitor is only getting 60Hz, and `sensors` doesn't even show my GPU anymore.  I don't know how because all I was doing was reading files and using apt to install packages (and then uninstall them when they made no changes.) Even last night my GPU was performing really terribly and I was not sure why. I thought rebooting today would help but it only got worse. 

Unfortunately I've lost track of exactly what changes I might have made in the past few days. I have no idea how I might have uninstalled my GPU, and the work I have been able to do to get it working was mostly cobbled together from tips online, so I'm not really sure how to put it back. I need to step back and figure out how to get back to square one.
